### PR TITLE
Added alder trees and a extractable basic hemostatic

### PR DESCRIPTION
**Pull request by MichaelMacha**
#### Summary

SUMMARY: Content "Added alder trees to the game. (Note that I have not _yet_ added it to any map regions; but it tends to thrive in cool, wet areas, usually near water.) I also added a new recipe for hemostatic powder involving crushed alder bark, which is generally harvestable in spring and autumn."

#### Purpose of change

There's reason to believe that chemistry skills could develop from basic cooking and first aid skills, given that proper botanical medicines are introduced into the game. While it's hardly the only use for alder, the crushed bark is hemostatic if you can get it fine enough, and having a recipe for hemostatic powder that does not depend on giant insects sounds pertinent.

#### Describe the solution

With a mortar and pestle, bark can be harvested from alder trees, combined with a little lye, and refined into hemostatic powder. The proportions could be looked over, but drying it out and crushing it should provide a quick stand-in for a bandage.

#### Describe alternatives you've considered

The only thing I'm not certain about is the lye, but it seemed more reasonable than having instant hemostatic powder simply because you've got a mortar and some trees around. It's also reasonable to make this a separate powder type of its own, rather than equating it to the commercial stuff, but I'm still familiarizing myself with the JSON format and it seemed wiser to take it a step at a time.

#### Testing

I spawned the products in in debug mode, and confirmed that they loaded properly and worked.

#### Additional context

This is currently an autolearned recipe; it probably shouldn't be. As I catch up with more local flora in the New England area, perhaps I can eventually add an encyclopedic reference or survival book with guides to creating these.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
Excellent, this gets you started right away.

We still have to add this to regional settings to get it to spawn. Check out this file: [regional map settings](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/regional_map_settings.json). It's got a number of spots we could make alders start appearing. I'll be able to help you look at those in the morning to offer suggestions.

**Comment by Maleclypse** [comment]
When I have failed JSON validation tests I copy the failed file in to https://dev.narc.ro/cataclysm/format.html.  IF that returns a syntax error I then goto https://jsonlint.com/ and work my way through until it is valid json then lint it again in the first page.

**Comment by andrei8l** [comment]
@Maleclypse the in-tree tool `json_formatter.cgi` should do the same thing. Report bugs if it doesn't :)

**Comment by Maleclypse** [comment]
> @Maleclypse the in-tree tool `json_formatter.cgi` should do the same thing. Report bugs if it doesn't :)

I will learn to use this.  Thanks!

**Comment by anothersimulacrum** [comment]
It's the same program dev.narc.ro uses, except the error reporting is worse on that (this tells you what syntax errors there are).
I'm not sure you can use .cgi on windows, though, and you need to compile it last I checked.

File: data/json/regional_map_settings.json [hunk 1]
```diff
@@ -549,7 +553,33 @@
...
+          "mx_sewer_pipesC": 30,
+          "mx_sewer_pipesD": 30,
+          "mx_sewer_pipesE": 30,
+          "mx_sewer_pipesF": 30,
+          "mx_sewer_pipesG": 30,
+          "mx_sewer_pipesH": 30,
+          "mx_sewer_pipesI": 30,
+          "mx_sewer_pipesJ": 30
+        }
+      }
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json:582]:
```suggestion
     "lake_shore": { "chance": 2, "extras": { "mx_reed": 100 }
```

File: data/json/regional_map_settings.json [hunk 2]
```diff
@@ -665,7 +695,15 @@
         "urban_5_house": 50,
         "urban_6_house": 50,
         "urban_7_house_garden": 50,
+        "urban_8_house_brick_garden": 50,
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json:698]:
```suggestion
```

File: data/json/regional_map_settings.json [hunk 3]
```diff
@@ -665,7 +695,15 @@
...
         "urban_9_house_garage_loft": 50,
+        "urban_10_house_brick_pool": 50,
+        "urban_11_house_brick": 50,
+        "urban_12_house": 50,
+        "urban_15_house": 50,
+        "urban_16_house_ranch": 50,
+        "urban_17_house_ranch": 50,
+        "urban_18_victorian": 50,
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json:706]:
```suggestion
```

File: data/json/regional_map_settings.json [hunk 4]
```diff
@@ -919,7 +957,6 @@
       "base_temperature": 6.5,
       "base_humidity": 70.0,
       "base_pressure": 1015.0,
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json:959]:
```suggestion
      "base_pressure": 1015.0,
      "base_acid": 0.0,
```

File: data/json/regional_map_settings.json [hunk 5]
```diff
@@ -932,8 +969,6 @@
         "rain",
         "thunder",
         "lightning",
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json:971]:
```suggestion
        "lightning",
        "acid_drizzle",
        "acid_rain",
```

File: data/json/regional_map_settings.json [hunk 6]
```diff
@@ -549,7 +553,7 @@
         }
       },
       "river": { "chance": 3, "extras": { "mx_reed": 100 } },
-      "lake_shore": { "chance": 2, "extras": { "mx_reed": 100 } }
+     "lake_shore": { "chance": 2, "extras": { "mx_reed": 100 }
```
Comment (ZhilkinSerg) [inline data/json/regional_map_settings.json:556]:
```suggestion
      "lake_shore": { "chance": 2, "extras": { "mx_reed": 100 } }
```

**Comment by ZhilkinSerg** [comment]
> We still have to add this to regional settings to get it to spawn. Check out this file: [regional map settings](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/regional_map_settings.json). It's got a number of spots we could make alders start appearing. I'll be able to help you look at those in the morning to offer suggestions.

I've added alder trees to regional settings with some arbitrary numbers, which we should review and update.

**Pull request merged**